modules/lib/lib-node/src/main/resources/lib/xp/node.ts

```diff
@@ -124,22 +126,22 @@ export interface SuggestionResult {
     }[];
 }
 
-export interface NodeQueryResult {
+export interface NodeQueryResult<AggregationOutput extends Record<string, AggregationsResult>> {
     total: number;
     count: number;
     hits: NodeQueryResultHit[];
-    aggregations?: Record<string, AggregationsResult>;
+    aggregations?: AggregationOutput;
     suggestions?: Record<string, SuggestionResult[]>;
 }
 
-export interface NodeMultiRepoQueryResult {
+export interface NodeMultiRepoQueryResult<AggregationOutput extends Record<string, AggregationsResult>> {
     total: number;
     count: number;
     hits: (NodeQueryResultHit & {
         repoId: string;
         branch: string;
     })[];
-    aggregations?: Record<string, AggregationsResult>;
+    aggregations?: AggregationOutput;
```

Comment:
Same as above. Should probably either be `never` or it should exist.

```suggestion
    aggregations: AggregationOutput;
```